ingenialink/dictionary.py

```diff
@@ -51,7 +112,153 @@ def __init__(self, dictionary_path):
         """Categories: Instance of all the categories in the dictionary."""
         self.errors = None
         """Errors: Instance of all the errors in the dictionary."""
+        self._registers = []
+        """Registers: Instance of all the registers in the dictionary"""
 
-    @abstractmethod
     def registers(self, subnode):
-        raise NotImplementedError
+        """Gets the register dictionary to the targeted subnode.
+
+        Args:
+            subnode (int): Identifier for the subnode.
+
+        Returns:
+            dict: Dictionary of all the registers for a subnode.
+
+        """
+        return self._registers[subnode]
+
+    def read_dictionary(self):
+        """Reads the dictionary file and initializes all its components."""
+        try:
+            with open(self.path, 'r', encoding='utf-8') as xml_file:
+                tree = ET.parse(xml_file)
+        except FileNotFoundError:
+            raise FileNotFoundError(f"There is not any xml file in the path: {self.path}")
+        root = tree.getroot()
+
+        device = root.find('./Body/Device')
```

Comment:
In my opinion, these string values should be constants, for example:

DICTIONARY_ROOT = "/IngeniaDictionary"
DICTIONARY_BODY = f"{DICTIONARY_ROOT}/Body"
DICTIONARY_DEVICE f"{DICTIONARY_BODY}/Device"
etc...